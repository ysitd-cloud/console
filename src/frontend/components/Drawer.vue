<template>
    <v-navigation-drawer v-model="drawerToggle" persistent light app>
        <v-toolbar flat class="transparent">
            <v-list class="pa-0">
                <v-list-tile href="https://account.ysitd.cloud/" target="_blank">
                    <v-list-tile-avatar>
                        <img :src="photo" />
                    </v-list-tile-avatar>
                    <v-list-tile-content>
                        <v-list-tile-title>{{ displayName }}</v-list-tile-title>
                    </v-list-tile-content>
                </v-list-tile>
            </v-list>
        </v-toolbar>
        <v-list>
            <template v-for="link in links">
                <v-list-tile v-if="link.external" :key="link.href" :href="link.href" target="_blank">
                    <span>{{ link.display }}</span>
                    <v-spacer />
                    <v-icon>launch</v-icon>
                </v-list-tile>
            </template>
            <v-list-tile href="/auth/logout">Logout</v-list-tile>
            <v-divider></v-divider>
        </v-list>
    </v-navigation-drawer>
</template>

<style scoped lang="stylus">
    .list__tile--link .icon
        font-size: 1rem
</style>

<script>
    import { mapState, mapGetters } from 'vuex';

    export default {
      props: {
        drawer: {
          type: Boolean,
          requierd: true,
        },
      },
      computed: {
        drawerToggle: {
          get() {
            return this.drawer;
          },
          set(value) {
            this.$emit('update:drawer', value);
          },
        },
        ...mapState({
          displayName: state => state.user.displayName,
        }),
        ...mapGetters({
          photo: 'user/photo',
        }),
      },
      data() {
        return {
          links: [
            { href: 'https://account.ysitd.cloud/', external: true, display: 'Account' },
            { href: 'https://health.ysitd.cloud/', external: true, display: 'Service Health' },
          ],
        };
      },
    };
</script>
