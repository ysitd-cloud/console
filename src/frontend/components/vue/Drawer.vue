<template>
    <v-navigation-drawer v-model="drawerToggle" fixed light app>
        <drawer-toolbar :photo="photo" :displayName="displayName"></drawer-toolbar>
        <v-list>
            <template v-for="link in links">
                <v-list-tile v-if="link.external" :key="link.href" :href="link.href" target="_blank">
                    <span>{{ link.display }}</span>
                    <v-spacer />
                    <v-icon>launch</v-icon>
                </v-list-tile>
                <v-list-tile v-else :key="link.display" :to="link.to">
                    <span>{{ link.display }}</span>
                </v-list-tile>
            </template>
            <v-divider></v-divider>
            <v-list-tile href="/auth/logout">Logout</v-list-tile>
        </v-list>
    </v-navigation-drawer>
</template>
<script>
    import { mapState, mapGetters } from 'vuex';
    import DrawerToolbar from './DrawerToolbar.vue';

    export default {
      components: {
        DrawerToolbar,
      },
      props: {
        drawer: {
          type: Boolean,
          requierd: true,
        },
      },
      computed: {
        drawerToggle: {
          get() {
            return this.drawer;
          },
          set(value) {
            this.$emit('update:drawer', value);
          },
        },
        ...mapState({
          displayName: state => state.user.displayName,
        }),
        ...mapGetters({
          photo: 'user/photo',
        }),
      },
      data() {
        return {
          links: [
            { to: { name: 'account.get' }, display: 'Account' },
            { href: 'https://health.ysitd.cloud/', external: true, display: 'Service Health' },
            { to: { name: 'app.list' }, display: 'App' },
          ],
        };
      },
    };
</script>
