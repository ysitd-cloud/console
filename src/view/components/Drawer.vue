<template>
    <v-navigation-drawer v-model="drawerToggle" persistent light>
        <v-toolbar flat class="transparent">
            <v-list class="pa-0">
                <v-list-tile>

                </v-list-tile>
            </v-list>
        </v-toolbar>
    </v-navigation-drawer>
</template>

<script>
    export default {
      props: {
        drawer: {
          type: Boolean,
          requierd: true,
        },
      },
      computed: {
        drawerToggle: {
          get() {
            return this.drawer;
          },
          set(value) {
            this.$emit('update:drawer', value);
          },
        },
      },
    };
</script>
