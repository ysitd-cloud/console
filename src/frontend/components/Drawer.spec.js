import { shallow, mount, createLocalVue } from '@vue/test-utils';
import { createRenderer } from 'vue-server-renderer';
import Vuex, { Store } from 'vuex';

import Drawer from './Drawer.vue';

/* globals describe, it, expect, jest, beforeEach */

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Component DrawerToolbar', () => {
  let store;
  let photoGetter;

  beforeEach(() => {
    photoGetter = jest.fn();
    store = new Store({
      modules: {
        user: {
          namespaced: true,
          state: {
            displayName: 'foo',
          },
          getters: {
            photo: photoGetter,
          },
        },
      },
    });
  });

  it('is a Vue instance', () => {
    const wrapper = mount(Drawer, {
      store,
      localVue,
      propsData: { drawer: true },
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('matches snapshot for display', () => {
    const renderer = createRenderer();
    const wrapper = shallow(Drawer, {
      store,
      localVue,
      propsData: { drawer: true },
    });
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
