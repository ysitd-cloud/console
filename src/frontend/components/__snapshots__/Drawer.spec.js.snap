// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component DrawerToolbar matches snapshot for display 1`] = `
<v-navigation-drawer fixed="" light="" app="">
    <!---->
    <v-list>
        <v-list-tile href="https://account.ysitd.cloud/" target="_blank"><span>Account</span>
            <v-spacer></v-spacer>
            <v-icon>launch</v-icon>
        </v-list-tile>
        <v-list-tile href="https://health.ysitd.cloud/" target="_blank"><span>Service Health</span>
            <v-spacer></v-spacer>
            <v-icon>launch</v-icon>
        </v-list-tile>
        <v-list-tile to="[object Object]"><span>App</span></v-list-tile>
        <v-divider></v-divider>
        <v-list-tile href="/auth/logout">Logout</v-list-tile>
    </v-list>
</v-navigation-drawer>
`;
